vizro-core/src/vizro/models/_components/form/_form_utils.py

```diff
@@ -94,5 +94,8 @@ def validate_step(cls, step, values):
     return step
 
 
-def set_default_marks(cls, v, values):
-    return v if values.get("step") is None else {}
+def set_default_marks(cls, marks, values):
+    if marks is None or (not marks and values.get("step") is None):
+        return None
+
+    return marks if values.get("step") is None else {}
```

Comment:
Can you check if this simplified version works fully:

```suggestion
def set_default_marks(cls, marks, values):
    if values.get("step") is None:
        marks = None
    return marks
```

Reply:
- what's the expected behavior when step is None and marks is not None?
- what's the expected behavior when step and marks are not None, but they are conflicting? Shall we suggest in the user guide which one takes precedence?